src/components/SemesterView.vue

```diff
@@ -107,6 +121,40 @@ export default {
         this.$emit('compact-updated', !this.compact);
       }
     },
+    buildDuplicateCautions() {
+      if (this.semesters) {
+        const coursesMap = {};
+        this.semesters.forEach(semester => {
+          semester.courses.forEach(course => {
+            if (coursesMap[`${course.subject} ${course.number}`]) course.alerts.caution = 'Duplicate';
+            coursesMap[`${course.subject} ${course.number}`] = true;
+          });
+        });
+      }
+    },
+    openSemesterConfirmationModal(type, year, isAdd) {
+      if (isAdd) {
+        this.confirmationText = `Added ${type} ${year} to plan`;
+      } else {
+        this.confirmationText = `Deleted ${type} ${year} from plan`;
+      }
+
+      const confirmationModal = document.getElementById(`semesterConfirmation`);
```

Comment:
This file will need to be heavily edited to remove all of the DOM manipulation.

Reply:
okie dokie willie